File: app/src/main/java/com/etesync/syncadapter/utils/HintManager.java
package com.etesync.syncadapter.utils;


import android.content.Context;
import android.content.SharedPreferences;

import java.util.LinkedList;
import java.util.List;

public class HintManager {
    private final static String PREF_NAME = "hints";

    private static SharedPreferences getPrefs(Context context) {
        return context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    }

    public static void setHintSeen(Context context, String hint, boolean seen) {
        SharedPreferences prefs = getPrefs(context);

        prefs.edit().putBoolean(hint, seen).apply();
    }

    public static boolean getHintSeen(Context context, String hint) {
        SharedPreferences prefs = getPrefs(context);

        return prefs.getBoolean(hint, false);
    }

    public static void resetHints(Context context) {
        SharedPreferences prefs = getPrefs(context);

        prefs.edit().clear().apply();
    }
}

File: app/src/main/java/com/etesync/syncadapter/utils/HintManager.kt
package com.etesync.syncadapter.utils


import android.content.Context
import android.content.SharedPreferences

object HintManager {
    private val PREF_NAME = "hints"

    private fun getPrefs(context: Context): SharedPreferences {
        return context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    }

    fun setHintSeen(context: Context, hint: String, seen: Boolean) {
        val prefs = getPrefs(context)

        prefs.edit().putBoolean(hint, seen).apply()
    }

    fun getHintSeen(context: Context, hint: String): Boolean {
        val prefs = getPrefs(context)

        return prefs.getBoolean(hint, false)
    }

    fun resetHints(context: Context) {
        val prefs = getPrefs(context)

        prefs.edit().clear().apply()
    }
}
